Assets/Capstone/Scripts/Environmental/Sun.cs

```diff
@@ -46,69 +67,149 @@ void Update()
+        //is Dusk
+        else if (angleInDegrees > duskInterval && angleInDegrees < nightInterval)
         {
-            sun.color = Color.Lerp(sun.color, night, Time.deltaTime / 10);
-            RenderSettings.ambientLight = Color.Lerp(RenderSettings.ambientLight, ambientNight, Time.deltaTime / 10);
-            //sun.intensity = Mathf.Lerp (sun.intensity, 2, Time.deltaTime);
-            isMidday = false;
-            isMorning = false;
-            isDusk = true;
-            isNight = false;
+            //lighting
+            LerpLighting(dusk, ambientDusk, skyDusk);
+            //time bool
+            SwitchTimeState(2);
         }
-        else if (transform.position.x < nightInterval)
+        //is Night
+        else if (angleInDegrees > nightInterval && angleInDegrees < morningInterval)
         {
-            isMorning = false;
-            isMidday = false;
-            isDusk = false;
-            isNight = true;
-            //waterDay.SetActive(false);
-            //waterNight.SetActive(true);
-            
+            //lighting
+            LerpLighting(night, ambientNight, skyNight);
+            //time bool
+            SwitchTimeState(3);
+
             //when its night, yesterday catches up to dayCounter
             yesterday = dayCounter;
+
+            //turn on stars
+            if (!stars.activeSelf )
+            {
+                stars.SetActive(true);
+            }
+        }
+    }
+
+    //pass the state you want to set true
+    void SwitchTimeState(int state)
+    {
+        switch (state)
+        {
+            //morning
+            case 0:
+                isMorning = true;
+                isMidday = false;
+                isDusk = false;
+                isNight = false;
+                break;
+            //midday
+            case 1:
+                isMorning = false;
+                isMidday = true;
+                isDusk = false;
+                isNight = false;
+                break;
+            //dusk
+            case 2:
+                isMorning = false;
+                isMidday = false;
+                isDusk = true;
+                isNight = false;
+                break;
+            //night
+            case 3:
+                isMorning = false;
+                isMidday = false;
+                isDusk = false;
+                isNight = true;
+                break;
         }
+    }
 
-        transform.LookAt(Vector3.zero);
+    //lerps lighting color values
+    void LerpLighting(Color sunC, Color ambientC, Material skyboxC)
+    {
+        sun.color = Color.Lerp(sun.color, sunC, Time.deltaTime / 10);
+        RenderSettings.ambientLight = Color.Lerp(RenderSettings.ambientLight, ambientC, Time.deltaTime / 10);
+        RenderSettings.skybox = skyboxC;
+    }
 
-   
+    //called every morning
+    void NewDay()
+    {
+        //when its morning increase dayCounter
+        if (dayCounter == yesterday)

```

Comment:
what is the rational with this if statement? Is NewDay being called more than once per day?